src/ColorSchemePreprocess.php

```diff
@@ -16,19 +17,58 @@ class ColorSchemePreprocess {
    *
    * @param array $variables
    *   The render array.
-   * @param bool $background
-   *   Add a background class if TRUE.
+   * @param array $options
+   *   An associative array of options to control the behavior of the function.
+   *   Possible keys are:
+   *   - 'text_colored': (bool) If set to TRUE, adds a class to override the
+   *     text color.
+   *   - 'background': (bool) If set to TRUE, adds a class to override the
+   *     background color.
+   *   - 'primary_background': (bool) If set to TRUE, adds a class to override
+   *     the background and text color to primary.
+   *   - 'secondary_background': (bool) If set to TRUE, adds a class to override
+   *     the background and text color to secondary.
+   *   - 'danger_background': (bool) If set to TRUE, adds a class to override
+   *     the background and text color to danger.
+   *   - 'success_background': (bool) If set to TRUE, adds a class to override
+   *     the background and text color to success.
    */
-  public function injectColorScheme(array &$variables, bool $background = TRUE): void {
+  public function injectColorScheme(array &$variables, array $options = []): void {
     if (!isset($variables['elements']['#oe_color_scheme'])) {
       return;
     }
 
-    $variables['attributes']['class'][] = Html::getClass($variables['elements']['#oe_color_scheme']);
-    $variables['attributes']['class'][] = 'text-color-default';
+    $variables['attributes'] = new Attribute($variables['attributes'] ?? []);
+    $variables['attributes']->addClass(Html::getClass($variables['elements']['#oe_color_scheme']));
 
-    if ($background) {
-      $variables['attributes']['class'][] = 'bg-default';
+    if (!empty($options['text_colored'])) {
```

Comment:
would like to make sure that its bool, could do `(isset && === TRUE)` but like this the complexity is lower. Was contemplating also the symfony options resolver but seems like bit much for a preprocess

Reply:
I have some second thoughts here:

1. If somebody makes a typo in the option id, they will never know, there is no validation. Maybe it would be good to introduce constants for options (like `const PRIMARY_BACKGROUND = 'primary_background`), or an `enum`, or defining a list of allowed options and throwing an `InvalidArgumentException`
2. Using `empty` is discouraged by the latest trends, and if we ever introduce PHPStan to the pipeline, it will start to throw errors. Perhaps `TRUE === $options['text_colored'] ?? FALSE` would be more readable? Or maybe we should define an array of default options, so they are always set.